REVIEWING.md

```diff
@@ -0,0 +1,84 @@
+# Reviewing pull requests
+
+This document provides a high-level and **incomplete** checklist of things that pull request reviewers should be aware of when performing code review. The checklist mainly consists of things that are specific to the scikit-bio project and that generally apply to incoming pull requests. The checklist is incomplete because it is not possible to describe all things to verify during code review (that depends on what is being reviewed). This document also doesn't attempt to describe *how* to perform code review (there are many online resources for that).
+
+Reviewers are encouraged to keep this document up-to-date as the project evolves and to add anything that's missing.
+
+The checklist is not in any particular order.
+
+## Changelog
+
+This is one of the easiest parts to forget in a pull request.
+
+- Note all public (i.e. user-facing) changes in `CHANGELOG.md` under the latest development version section at the top of the file. This includes things like bug fixes, API additions/changes/removal, performance enhancements, etc. The changelog has several subsections for organizing these changes.
+- If a corresponding issue exists, it should be linked to from the changelog.
+- Use public imports (`skbio.sequence.Sequence` instead of `skbio.sequence._sequence.Sequence`) when documenting import paths in the changelog.
+- Internal changes not visible/applicable to users (e.g. refactoring, private methods, etc.) are better suited for commit messages than the changelog.
+
+## Public vs. private API
+
+Be aware of what type of API changes are being made. scikit-bio uses the following conventions to distinguish public vs. private API:
+
+**Public API:** API with an import path that doesn't contain leading underscores in any submodule/subpackage/object names. Users and scikit-bio devs can use public API. Example: `skbio.sequence.Sequence`, `skbio.stats.subsample`
+
+**Package-private API:** API with an import path containing a leading underscore in at least one submodule/subpackage. Users should not import package-private API. Package-private API can be imported anywhere *within* the scikit-bio package. Example: `skbio.util._misc.chunk_str`, `skbio.util._testing.ReallyEqualMixin`
+
+**Private API:** API with object name starting with an underscore. Users should not import private API. Private API should only be imported and used in the module where it is defined. It should not be imported in other parts of the scikit-bio package. Example: `skbio.util._testing._normalize_signs`, `skbio.stats.composition._gram_schmidt_basis`
+
```

Comment:
I think we should make clear here that moving forward, all private API should starts with underscore (no matter it is package private API or true private API), to avoid keeping track if an API is imported to subpackage as public or not. So that we can immediately know by the API name if it is private, as we discussed.